src/pipeline.cc

```diff
@@ -892,6 +892,14 @@ class PipelineWorker : public Napi::AsyncWorker {
           image.set(s.first.data(), s.second.data());
         }
       }
+      // XMP buffer
+      if (baton->withXmpBuffer != nullptr && baton->withXmpBufferLength > 0) {
+        image = image.copy();
```

Comment:
very very small nit here since there is the same for exif, but I don't understand why we copy the image here 🤔